ASM/c/pots.c

```diff
@@ -2,72 +2,66 @@
 #include "n64.h"
 #include "gfx.h"
 
-#define BROWN_FRONT_TEXTURE 0x06000000
+#define POT_FRONT_TEXTURE 0x06000000
+#define POT_FRONT_DLIST 0x060017C0
 
 uint32_t POT_TEXTURE_MATCH_CONTENTS = 0;
 
-extern void* GILDED_CHEST_FRONT_TEXTURE;
-extern void* SILVER_CHEST_FRONT_TEXTURE;
-extern void* SKULL_CHEST_FRONT_TEXTURE;
+extern void *GILDED_CHEST_FRONT_TEXTURE;
+extern void *SILVER_CHEST_FRONT_TEXTURE;
+extern void *SKULL_CHEST_FRONT_TEXTURE;
 
-extern Mtx_t* write_matrix_stack_top(z64_gfx_t* gfx);
+extern Mtx_t *write_matrix_stack_top(z64_gfx_t *gfx);
 asm(".equ write_matrix_stack_top, 0x800AB900");
 
-void get_pot_override(z64_actor_t *actor) {
-    uint8_t size  = ((uint8_t*)actor)[0x01E9];
-    uint8_t color = size;
+override_t get_pot_override(z64_actor_t *actor, z64_game_t *game) {
+    // make a dummy EnItem00 with enough info to get the override
+    EnItem00 dummy;
+    dummy.collectibleFlag = (actor->variable & 0x3F00) >> 8;
+    dummy.actor.actor_id = 0x15;
+    dummy.actor.dropFlag = 1;
+    dummy.actor.dropFlag |= (actor->variable & 0x001F) >> 5;
```

Comment:
don't need this (it actually won't do anything as currently written). I add 2 additional bits of flag space to mask 0xC0 of the actor variable for en_item00s when dropped from things like crates. Pots don't currently use this.